Source/Core/VideoCommon/BPStructs.cpp

```diff
@@ -893,359 +892,302 @@ void GetBPRegInfo(const u8* data, std::string* name, std::string* desc)
-  {
-    SetRegName(BPMEM_BLENDMODE);
-    BlendMode mode;
-    mode.hex = cmddata;
-    *desc = fmt::format("Enable: {}\n"
-                        "Logic ops: {}\n"
-                        "Dither: {}\n"
-                        "Color write: {}\n"
-                        "Alpha write: {}\n"
-                        "Dest factor: {}\n"
-                        "Source factor: {}\n"
-                        "Subtract: {}\n"
-                        "Logic mode: {}\n",
-                        no_yes[mode.blendenable], no_yes[mode.logicopenable], no_yes[mode.dither],
-                        no_yes[mode.colorupdate], no_yes[mode.alphaupdate], mode.dstfactor,
-                        mode.srcfactor, no_yes[mode.subtract], mode.logicmode);
-  }
-  break;
+    return std::make_pair(RegName(BPMEM_BLENDMODE),
+                          fmt::format("Blend mode: {}", BlendMode{.hex = cmddata}));
 
   case BPMEM_CONSTANTALPHA:  // 0x42
-    SetRegName(BPMEM_CONSTANTALPHA);
-    // TODO: Description
-    break;
+    return std::make_pair(RegName(BPMEM_CONSTANTALPHA),
+                          fmt::format("Constant alpha: {}", ConstantAlpha{.hex = cmddata}));
 
   case BPMEM_ZCOMPARE:  // 0x43
-  {
-    SetRegName(BPMEM_ZCOMPARE);
-    PEControl config;
-    config.hex = cmddata;
-    *desc = fmt::format("EFB pixel format: {}\n"
-                        "Depth format: {}\n"
-                        "Early depth test: {}\n",
-                        config.pixel_format, config.zformat, no_yes[config.early_ztest]);
-  }
-  break;
+    return std::make_pair(RegName(BPMEM_ZCOMPARE),
+                          fmt::format("Z compare: {}", PEControl{.hex = cmddata}));
 
   case BPMEM_FIELDMASK:  // 0x44
-    SetRegName(BPMEM_FIELDMASK);
-    // TODO: Description
-    break;
+    return std::make_pair(RegName(BPMEM_FIELDMASK),
+                          fmt::format("Field mask: {}", FieldMask{.hex = cmddata}));
 
   case BPMEM_SETDRAWDONE:  // 0x45
-    SetRegName(BPMEM_SETDRAWDONE);
+    return DescriptionlessReg(BPMEM_SETDRAWDONE);
     // TODO: Description
-    break;
 
   case BPMEM_BUSCLOCK0:  // 0x46
-    SetRegName(BPMEM_BUSCLOCK0);
+    return DescriptionlessReg(BPMEM_BUSCLOCK0);
     // TODO: Description
-    break;
 
   case BPMEM_PE_TOKEN_ID:  // 0x47
-    SetRegName(BPMEM_PE_TOKEN_ID);
+    return DescriptionlessReg(BPMEM_PE_TOKEN_ID);
     // TODO: Description
-    break;
 
   case BPMEM_PE_TOKEN_INT_ID:  // 0x48
-    SetRegName(BPMEM_PE_TOKEN_INT_ID);
+    return DescriptionlessReg(BPMEM_PE_TOKEN_INT_ID);
     // TODO: Description
-    break;
 
   case BPMEM_EFB_TL:  // 0x49
   {
-    SetRegName(BPMEM_EFB_TL);
     X10Y10 left_top;
     left_top.hex = cmddata;
-    *desc = fmt::format("Left: {}\nTop: {}", u32(left_top.x), u32(left_top.y));
+    return std::make_pair(RegName(BPMEM_EFB_TL), fmt::format("EFB Left: {}\nEFB Top: {}",
+                                                             u32(left_top.x), u32(left_top.y)));
```

Comment:
The `u32(left_top.x)`-type casts are no longer needed as `BitField` has a formatter for it (they were needed when using raw bitfields though).  Also, it can be made `const` using `const X10Y10 left_top{.hex = cmddata}`.